client/src/components/signup/Signup.js

```diff
@@ -14,26 +19,50 @@ function Signup() {
   // state formChoice variable accordingly
   const handleFormChoiceClick = (choice) => (setFormChoice(choice))
 
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const {
+    isFetching, isSuccess, isError, errorMessage,
+  } = useSelector(userSelector)
+
   // onValidateSubmit function is a function that is being passed as prop
   // to the SignupForm component. This function gathers the data from the form
   // component and comes back to this Signup component with validated credentials
   // that are ready to be passed onto the Login or Signup functions.
   const onValidatedSubmit = (validatedCredentials) => {
     if (formChoice === 'login') {
-      console.log('login with: ', validatedCredentials)
+      dispatch(signupUser(validatedCredentials))
     }
     if (formChoice === 'signup') {
-      console.log('signup with: ', validatedCredentials)
+      dispatch(loginUser(validatedCredentials))
```

Comment:
I have not seen the login action you created so I cannot tell what the specification is. But I recommend you do your validation first, then after that call dispatch and dispatch the login action.

Reply:
I commented login out for now and I'm doing solely signup at the moment. This was only a non-working example of how I saw the Signup.js in the final state.